src/client/Client.cc

```diff
@@ -5440,8 +5440,16 @@ int Client::may_setattr(Inode *in, struct ceph_statx *stx, int mask,
   }
 
   if (mask & CEPH_SETATTR_MODE) {
-    if (perms.uid() != 0 && perms.uid() != in->uid)
-      goto out;
+    if (perms.uid() != 0 && perms.uid() != in->uid) {
+      if(in->caps_issued_mask(CEPH_CAP_AUTH_EXCL)) {
```

Comment:
why do you check CEPH_CAP_AUTH_EXCL? It's possible that multiple clients open a setuid file for write at the same time.

Reply:
This place is wrong, I modified it, please review again.
I use the pjdfstest  tool for a test, the compatibility result is correct